src/smbclient/_pool.py

```diff
@@ -99,9 +98,9 @@ def __init__(
         self.skip_dfs = skip_dfs
         self.auth_protocol = auth_protocol
         self.require_secure_negotiate = require_secure_negotiate
-        self._domain_controller = domain_controller  # type: Optional[str]
-        self._domain_cache = []  # type: List[DomainEntry]
-        self._referral_cache = []  # type: List[ReferralEntry]
+        self._domain_controller: str | None = domain_controller
```

Comment:
I've added `from __future__ import annotations` to the beginning of the file. This allows to use modern annotations syntax in old Python versions, like 3.7, by converting all type hints to string representation. So this new syntax does not affect runtime, as you can see, tests are not failing on Python 3.7.

Tools like mypy can parse type hints to get proper types.

Reply:
Thanks for the clarification, I always assumed mypy was still dependent on what Python actually supported. Good to know I can use the newer syntax because it is much nicer.